spyder/plugins/projects/widgets/projectdialog.py

```diff
@@ -147,6 +149,8 @@ def _validate_location(
             reasons["location_not_writable"] = True
         elif name is not None:
```

Comment:
```suggestion
        elif os.name == "nt" and re.search(r":", location):
            # Prevent creating a project in directory with colons.
            # Fixes spyder-ide/spyder#16942
            reasons["wrong_name"] = True
        elif name is not None:
```

This will prevent the error reported on issue #16942 for existing directories.